modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/domain/DefaultMuleDomain.java

```diff
@@ -220,7 +222,17 @@ public void stop() {
         logger.info(miniSplash(format("Stopping domain '%s'", getArtifactName())));
       }
       if (this.artifactContext != null) {
-        this.artifactContext.getMuleContext().stop();
+        Reference<MuleException> error = new Reference<>();
+        withContextClassLoader(deploymentClassLoader.getClassLoader(), () -> {
+          try {
+            artifactContext.getMuleContext().stop();
+          } catch (MuleException e) {
+            error.set(e);
+          }
+        });
+        if (error.get() != null) {
+          throw error.get();
```

Comment:
do you need this? `withContextClassLoader` already wraps and throws any exception